model/converter/thrift/zipkin/to_domain.go

```diff
@@ -128,7 +152,25 @@ func (td toDomain) transformSpan(zSpan *zipkincore.Span) *model.Span {
 		Duration:      model.MicrosecondsAsDuration(uint64(zSpan.GetDuration())),
 		Tags:          tags,
 		Logs:          td.getLogs(zSpan.Annotations),
+	}}
+
+	if td.isClientAndServerSpan(zSpan) {
+		// if the span is client and server we split it into two separate spans
+		s := &model.Span{
+			TraceID:       model.TraceID{High: uint64(traceIDHigh), Low: uint64(zSpan.TraceID)},
+			SpanID:        model.SpanID(zSpan.ID),
+			OperationName: zSpan.Name,
+			ParentSpanID:  model.SpanID(parentID),
+		}
+		// if the first span is a client span we create server span and vice-versa.
+		if spanKindTag.VStr == string(ext.SpanKindRPCClient.Value.(ext.SpanKindEnum)) {
+			s.Tags = []model.KeyValue{model.String(ext.SpanKindRPCServer.Key, string(ext.SpanKindRPCServer.Value.(ext.SpanKindEnum)))}
```

Comment:
why cast the value `ext.SpanKindRPCServer.Value.(ext.SpanKindEnum)`?

Reply:
also, why is the first argument `SpanKindRPCServer` instead of just `SpanKind`?

Reply:
>why cast the value ext.SpanKindRPCServer.Value.(ext.SpanKindEnum)? 

because it fails on ` cannot convert ext.SpanKindRPCClient.Value (type interface {}) to type string`